src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = { ...state };
+  const combinedStates = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateClone, action.extraData);
+        break;
+
+      case 'removeProperties':
+        const { keysToRemove } = action;
+
+        for (const key of keysToRemove) {
+          if (stateClone.hasOwnProperty(key)) {
```

Comment:
This check is not obligatory here, cause If the property which you are trying to delete does not exist, delete will not have any effect and will return true. [(from MDN documentation)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete)